src/main/groovy/cruise/umple/UmpleGradlePlugin.groovy

```diff
@@ -164,22 +163,23 @@ class UmpleGradlePlugin implements Plugin<Project> {
         
         if (out.language.contains(UmpleLanguage.JAVA) && out.compileGenerated) 
         {      
-            project.tasks.getByName(sourceSet.compileJavaTaskName).dependsOn umpleGenerate
+            // The user has requested that generated Java files are compiled. Make sure
+            // the compileJava task runs after the compileUmple task
+            umpleGenerate.finalizedBy project.tasks.getByName(sourceSet.compileJavaTaskName)
```

Comment:
This is over using the umple compile mechanics? If so, I much prefer this -- use Gradle to it's advantage.

Reply:
Yeah, when it comes to compiling Java files we don't (and never have as far as I can tell) use the Umple mechanic. This change makes it so the user can set the `compileGenerated` flag and invoke the `compileUmple` task in order to generate and compile. Before the user had to use `compileJava` if they wanted to compile the generated source. It made the `compileGenerated` flag redundant.

Reply:
I think this should be on by default. I can't think of a reason to have it be off by default.